### PR TITLE
Use a promise for getDownloadUrl.

### PR DESCRIPTION
**Pull request by ian-r-rose**
Came across this when writing a `ContentsManager` for Google Drive. In that case the download url is not computable client-side from the path (it is instead based on a unique file Id hash), so in order to get the correct URL, there must be some communication with the server.
Cf. jupyterlab/services#297 and [https://github.com/ian-r-rose/jupyterlab_google_drive/tree/filebrowser]()

### CODE REVIEW

File: src/docregistry/context.ts [hunk 1]
```diff
@@ -354,7 +354,7 @@ class Context<T extends DocumentRegistry.IModel> implements DocumentRegistry.ICo
   /**
    * Resolve a relative url to a correct server path.
    */
-  resolveUrl(url: string): string {
+  resolveUrl(url: string): Promise<string> {
     // Ignore urls that have a protocol.
     if (utils.urlParse(url).protocol || url.indexOf('//') === 0) {
       return url;
```
Comment (afshin) [inline src/docregistry/context.ts]: `return url;` should be:
```
return Promise.resolve(url);
```

**Comment by ian-r-rose** [comment]
My mistake, ready for another look.

**Comment by ian-r-rose** [comment]
Okay, ready for another review. This wound up making a few more changes than I anticipated.

**Comment by blink1073** [comment]
@ian-r-rose, I just released `@jupyterlab/services@0.36.0` with your changes.

**Comment by ian-r-rose** [comment]
Awesome, thanks for sneaking it in under the wire.

**Comment by blink1073** [comment]
I hope you don't mean the JupyterLab 0.13 release, because we just did that...

**Comment by ian-r-rose** [comment]
No, should be fine.

**Comment by blink1073** [comment]
You should be able to `npm install --save @jupyterlab/services@latest` and update for the changes.

**Comment by blink1073** [comment]
Superseded by #1540.

**Pull request closed without merging**